ddl/ddl_worker.go

```diff
@@ -935,7 +935,12 @@ func updateSchemaVersion(t *meta.Meta, job *model.Job) (int64, error) {
 			}
 		}
 	case model.ActionAlterTableAlterPartition:
-		diff.TableID = job.CtxVars[0].(int64)
+		diff.TableID = job.TableID
+		diff.AffectedOpts = []*model.AffectedOption{
+			{
+				PartitionID: job.CtxVars[0].(int64),
```

Comment:
In `AffectedOption`, `TableID` and `OldTableID` can be taken as partition IDs.
See `ActionDropTablePartition`, `ActionTruncateTablePartition`, `ActionExchangeTablePartition` and other partition related operations.

Reply:
updated.